### PR TITLE
[processing] support for explicit output vector types

### PR DESCRIPTION
**Pull request by alexbruy**
Allow to set geometry type for output vectors explicitly. As result when chaining algorithms in modeler users will see only compatible layers.

### CODE REVIEW

File: python/plugins/processing/core/outputs.py [hunk 1]
```diff
@@ -256,13 +256,23 @@ def getTableWriter(self, fields):
 
 class OutputVector(Output):
 
+    VECTOR_TYPE_POINT = 0
+    VECTOR_TYPE_LINE = 1
+    VECTOR_TYPE_POLYGON = 2
+    VECTOR_TYPE_ANY = -1
```
Comment (arnaud-morvan) [inline python/plugins/processing/core/outputs.py]: This seems duplicated with ParameterVector, could be possible to share

File: python/plugins/processing/core/outputs.py [hunk 2]
```diff
@@ -336,3 +346,17 @@ def getVectorWriter(self, fields, geomType, crs, options=None):
...
+            if t == self.VECTOR_TYPE_POINT:
+                types += 'point, '
+            elif t == self.VECTOR_TYPE_LINE:
+                types += 'line, '
+            elif t == self.VECTOR_TYPE_POLYGON:
+                types += 'polygon, '
+            else:
+                types += 'any, '
+
+        return types[:-2]
```
Comment (arnaud-morvan) [inline python/plugins/processing/core/outputs.py]: This seems duplicated with ParameterVector, could be possible to share

File: python/plugins/processing/modeler/ModelerParametersDialog.py [hunk 3]
```diff
@@ -346,7 +349,13 @@ def getWidgetFromParameter(self, param):
...
             if param.datatype == ParameterMultipleInput.TYPE_VECTOR_ANY:
                 options = self.getAvailableValuesOfType(ParameterVector, OutputVector)
-            elif aram.datatype == ParameterMultipleInput.TYPE_RASTER:
+            elif param.datatype == ParameterMultipleInput.TYPE_VECTOR_POINT:
+                options = self.getAvailableValuesOfType(ParameterVector, None, [ParameterVector.TYPE_VECTOR_POINT])
+            elif param.datatype == ParameterMultipleInput.TYPE_VECTOR_LINE:
+                options = self.getAvailableValuesOfType(ParameterVector, None, [ParameterVector.TYPE_VECTOR_LINE])
+            elif param.datatype == ParameterMultipleInput.TYPE_VECTOR_POLYGON:
+                options = self.getAvailableValuesOfType(ParameterVector, None, [ParameterVector.TYPE_VECTOR_POLYGON])
+            elif param.datatype == ParameterMultipleInput.TYPE_RASTER:
```
Comment (arnaud-morvan) [inline python/plugins/processing/modeler/ModelerParametersDialog.py]: If constants were shared we could have something like : 
```
if param.datatype in ( point, line, polygon ):
    options = self.getAvailableValuesOfType(ParameterVector, None, param.datatype)
```

File: python/plugins/processing/core/outputs.py (hunk 1 above)
Comment (m-kuhn) [inline python/plugins/processing/core/outputs.py]: Or standard QGIS geometry types?
https://github.com/qgis/QGIS/blob/master/src/core/geometry/qgswkbtypes.h#L100-L107

File: python/plugins/processing/core/parameters.py [hunk 4]
```diff
@@ -1036,18 +1021,7 @@ def getFileFilter(self):
...
-                types += 'point, '
-            elif t == self.VECTOR_TYPE_LINE:
-                types += 'line, '
-            elif t == self.VECTOR_TYPE_POLYGON:
-                types += 'polygon, '
-            else:
-                types += 'any, '
-
-        return types[:-2]
+        return dataobjects.vectorDataType(self)
```
Comment (arnaud-morvan) [inline python/plugins/processing/core/parameters.py]: It should be better to have ``dataobjects.vectorDataType(self.datatype)``

**Pull request merged**